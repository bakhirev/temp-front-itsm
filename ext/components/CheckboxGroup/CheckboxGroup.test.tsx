import React from 'react';
import { shallow } from 'enzyme';

import { CheckboxGroup } from './';

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
}));

describe('CheckboxGroup', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const checkboxGroupProps = {
    list: [
      { id: 1, label: 'First item' },
      { id: 2, label: 'Second item' },
      { id: 3, label: 'Third item' },
    ],
    onChange: jest.fn(),
  };

  it('should render component with required properties', () => {
    const wrapper = shallow(<CheckboxGroup {...checkboxGroupProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label', () => {
    const wrapper = shallow(<CheckboxGroup {...checkboxGroupProps} label="CheckboxGroup" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small size', () => {
    const wrapper = shallow(<CheckboxGroup {...checkboxGroupProps} size="small" />);
    expect(wrapper).toMatchSnapshot();
  });
});
