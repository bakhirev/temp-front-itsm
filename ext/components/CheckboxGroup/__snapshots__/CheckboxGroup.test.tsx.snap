// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should render component with label 1`] = `
<styled.div>
  <Styled(styled.div)
    size="big"
    theme={Object {}}
  >
    CheckboxGroup
  </Styled(styled.div)>
  <styled.div
    key="0"
    size="big"
  >
    <Component
      checked={false}
      label="First item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
  <styled.div
    key="1"
    size="big"
  >
    <Component
      checked={false}
      label="Second item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
  <styled.div
    key="2"
    size="big"
  >
    <Component
      checked={false}
      label="Third item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
</styled.div>
`;

exports[`CheckboxGroup should render component with required properties 1`] = `
<styled.div>
  <styled.div
    key="0"
    size="big"
  >
    <Component
      checked={false}
      label="First item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
  <styled.div
    key="1"
    size="big"
  >
    <Component
      checked={false}
      label="Second item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
  <styled.div
    key="2"
    size="big"
  >
    <Component
      checked={false}
      label="Third item"
      onChange={[Function]}
      size="big"
    />
  </styled.div>
</styled.div>
`;

exports[`CheckboxGroup should render component with small size 1`] = `
<styled.div>
  <styled.div
    key="0"
    size="small"
  >
    <Component
      checked={false}
      label="First item"
      onChange={[Function]}
      size="small"
    />
  </styled.div>
  <styled.div
    key="1"
    size="small"
  >
    <Component
      checked={false}
      label="Second item"
      onChange={[Function]}
      size="small"
    />
  </styled.div>
  <styled.div
    key="2"
    size="small"
  >
    <Component
      checked={false}
      label="Third item"
      onChange={[Function]}
      size="small"
    />
  </styled.div>
</styled.div>
`;
